components/match2/commons/match_group_input.lua

```diff
@@ -250,7 +250,7 @@ function MatchGroupInput.readBracket(bracketId, args, options)
 	return matches, warnings
 end
 
----@param bracketData MatchGroupUtilBracketData
```

Comment:
What does this fix? Seems like a step back

Reply:
because it is not `MatchGroupUtilBracketData` it is "just" a table
it only gets made `MatchGroupUtilBracketData` by `MatchGroupUtil.bracketDataFromRecord`

basically
`MatchGroup.Bracket` calls `MatchGroupInput.readBracket` which calls `MatchGroupInput._readQualifiedHeader`
at this point none of the classes we have apply to the match table
then it gets stored and after that prepared for display and here it gets converted to the classes
and from that point onwards `match.bracketdata` is `MatchGroupUtilBracketData`